x-pack/plugins/observability/public/pages/alert_details/alert_details.tsx

```diff
@@ -62,10 +67,12 @@ export function AlertDetails() {
     ruleId: alert?.fields[ALERT_RULE_UUID],
   });
   const [summaryFields, setSummaryFields] = useState<AlertSummaryField[]>();
+  const [alertStatus, setAlertStatus] = useState<string>();
 
   useEffect(() => {
     if (alert) {
       setRuleTypeModel(ruleTypeRegistry.get(alert?.fields[ALERT_RULE_TYPE_ID]!));
+      setAlertStatus(alert?.fields[ALERT_STATUS]);
```

Comment:
I have refactored the code in this [commit](https://github.com/elastic/kibana/pull/174153/commits/416b995c83d9e1113343e92649ed6a1d78bdc391) to use `AlertStatus` wherever possible instead of `string`. I kept it as `string` in `const [alertStatus, setAlertStatus] = useState<string>();` because if I don't set the default state, I have to make either `alertStatus` in `HeaderActionsProps` and `PageTitleProps` optional or handle `undefined` state before passing props.

Reply:
Nice!
I still don't understand one scenario: what will happen at the start before having the alert status? The state value is `undefined,` right? So, how will the `AlertLifecycleStatusBadge` component handle this case?

Reply:
So, do you mean the current behaviour is similar to setting `ALERT_STATUS_ACTIVE` as the default value?
```
const [alertStatus, setAlertStatus] = useState<AlertStatus>(ALERT_STATUS_ACTIVE);
```

Then we don't need to cast `alertStatus` when passing `PageTitle` or `HeaderActions`.

Reply:
Right, but setting explicitly `ALERT_STATUS_ACTIVE` when the alert can be `recovered` or `untracked` could be confusing when reading code in future. From my point of view, keeping `undefined` is easy to understand that the status is set somewhere else.

Reply:
The difference is that the option that I proposed is more explicit; we already have this confusion behind the scenes; by looking at the code, it is unclear what status is set for the alert in case of an `undefined`.

Another option could be only to show the status if it exists, so we need to accommodate `undefined` if the alert has not been loaded yet.

Reply:
I like the idea of not showing status if alert is not loaded yet and alert status is therefore `undefined`. We can keep `alertStatus` as optional in`HeaderActionsProps` and`PageTitleProps` and show status badge only when `alertStatus` has status.

```
{
  alertStatus && <AlertLifecycleStatusBadge
    alertStatus={alertStatus}
    flapping={alert.fields[ALERT_FLAPPING]}
  />
}
```

Reply:
I like this idea, too :)